_uw-research-computing/hpc-spack-install.md

```diff
@@ -0,0 +1,391 @@
+
+Next, remember to reload the Spack environment
+
+```
+spack env activate yourEnvironmentName
+```
+{:.term}
+
+since the interactive session will start with the environment deactivated. 
+
+### ii. Create the local scratch directory
+
+Using the default configuration from [Setting Up Spack on HPC](hpc-spack-setup.md), Spack will try to use the machine's local disk space for staging and compiling files before transferring the finished results to the final installation directory.  Using this space will greatly improve the speed of the installation process. Create the local directory with the command
+
+```
+mkdir /local/yourNetID/spack_build
+```
+{:.term}
+
+where you should replace `yourNetID` with your NetID. At the end of the session, remember to delete this directory so that other people can use the disk space in their jobs. 
+
+> If the directory already exists, that means you forgot to remove it after one of your previous Spack installation sessions.  Simply remove the directory and make it again.
+>
+> ```
+> rm -rf /local/yourNetID/spack_build
+> mkdir /local/yourNetID/spack_build
+> ```
+> {:.term}
+
+### iii. Check the programs/packages to be installed
+
+If you've added the installation specifications to the environment, then you can check the installation plan using the command
+
+```
+spack spec -lI
+```
+{:.term}
+
+(the first letter after the hyphen is a lowercase "L" and the second letter is an uppercase "i").
+
+This command identifies what dependencies Spack needs in order to install your desired packages along with how it will obtain them.  Assuming their are no problems, then it will print a list of the packages and their dependencies, where entries that begin with a green `[+]` have already been installed somewhere in your local Spack installation, while those that begin with a green `[^]` are referencing the system installation, and those beginning with a gray `-` will need to be downloaded and installed.  
+
+If you are satisfied with the results, then you can proceed to install the programs.
+
+### iv. Install the environment packages
+
+Assuming that you are in an interactive Slurm session and have activated the desired environment containing the package specifications, you can run
+
+```
+spack install -j 4
+```
+{:.term}
+
+to install the packages inside of the Spack environment, where the number that comes after `-j` needs to match the number that you noted from when you started the interactive session (the one after `-n` when you ran the `srun` command for the interactive session). You can also add the `-v` option to have the installation be verbose, which will cause Spack to print the compile and make outputs in addition to the standard Spack output.
```

Comment:
This is in the order I would have people doing it. They should figure out what packages they want to install before they start an interactive session, because the interactive session is on a timer and compiling could take a while. This order reduces the chances of them timing out during the install.